src/main/java/de/tum/in/www1/artemis/service/metis/AnswerMessageService.java

```diff
@@ -77,10 +84,11 @@ public AnswerPost createAnswerMessage(Long courseId, AnswerPost answerMessage) {
         if (answerMessage.getId() != null) {
             throw new BadRequestAlertException("A new answer post cannot already have an ID", METIS_ANSWER_POST_ENTITY_NAME, "idexists");
         }
-        conversationService.isMemberElseThrow(answerMessage.getPost().getConversation().getId(), author.getId());
 
         Conversation conversation = conversationRepository.findByIdElseThrow(answerMessage.getPost().getConversation().getId());
 
+        conversationService.isMemberOrCreateForCourseWideElseThrow(conversation.getId(), author, Optional.empty());
```

Comment:
Why this is after getting object from db?

Reply:
Good catch, thank you. I check it before now.